packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -388,30 +388,66 @@ export class MirrorNodeClient {
         }
     }
 
+    private handleAccountNotFound(error: any, idOrAliasOrEvmAddress, requestIdPrefix) : null {
+        if(error instanceof MirrorNodeClientError) {
+            if (error.statusCode == 404) {
+                return null;
+
+            } else if (error.statusCode == 400) {
+                this.logger.debug(`${formatRequestIdMessage(requestIdPrefix)} Got Invalid Parameter when trying to fetch account from mirror node: ${JSON.stringify(error)}`);
+                throw predefined.INVALID_PARAMETER(idOrAliasOrEvmAddress, `Invalid 'address' field in transaction param. Address must be a valid 20 bytes hex string`);
+            }
+        }
+        // if is not a mirror node client error 400 or 404, rethrow
+        this.logger.error(`${formatRequestIdMessage(requestIdPrefix)} Unexpected error raised while fetching account from mirror-node: ${JSON.stringify(error)}`);
+        throw error;
+    }
+
     public async getAccount(idOrAliasOrEvmAddress: string, requestIdPrefix?: string) {
         return this.get(`${MirrorNodeClient.GET_ACCOUNTS_BY_ID_ENDPOINT}${idOrAliasOrEvmAddress}?transactions=false`,
             MirrorNodeClient.GET_ACCOUNTS_BY_ID_ENDPOINT,
             requestIdPrefix);
     }
 
-    public async getAccountLatestEthereumTransactionsByTimestamp(idOrAliasOrEvmAddress: string, timestampTo: string, numberOfTransactions: number = 1, requestIdPrefix?: string) {
+    /**
+     * Returns account or null if account does not exist
+     * @param {string} idOrAliasOrEvmAddress commonly an evm address
+     * @param {string} requestIdPrefix optional request id prefix
+     */
+    public async getAccountOrNull(idOrAliasOrEvmAddress: string, requestIdPrefix?: string) {
```

Comment:
Q: why the `OrNull` method overload?
Should services calling this handle the null?

Reply:
That was my original approach, but then realized that this methods are used by many different services (eth.ts, precheck.ts and relay.ts) and that I had to handle basically the same logic in all 3 services, so I needed to share the `handleAccountNotFound` logic in all 3 classes, so Instead of copy pasting (or creating a new helper class for the mirrorNodeClient), and then  modify all the reference where the logic already expects a `null` to add a try catch block, did not seem like progress or improvement to the code.

So I came up with this other approach, where the method name keeps working as expected returning null, but with a more explicit intention due to the `OrNull` overload and only handling `404` errors, not `400` or any other exception. and I get that this is eerly similar to the current `acceptedErrorStatusCodes` collection approach, but on this case the use of that error handler is intentional and specific to these methods since in these logic flows getting a `404` most be handled and is already handled by expecting the result to be null. (this has the advantage of re-using current logic flow where null are expected, not needing to add redundant try catch blocks every reference of the method, and re-utilizing the same error handler code logic for NotFound accounts)